daemon-scheduler/pkg/api/v1/api_test.go

```diff
@@ -64,13 +70,119 @@ func (suite *APITestSuite) TestPing() {
+
+	responseRecorder := httptest.NewRecorder()
+	suite.router.ServeHTTP(responseRecorder, request)
+
+	assert.Equal(suite.T(), http.StatusOK, responseRecorder.Code)
+
+	var environmentModel models.Environment
+	b, _ := ioutil.ReadAll(responseRecorder.Body)
+	json.Unmarshal(b, &environmentModel)
+
+	suite.assertSame(environment, &environmentModel)
+}
+
+func (suite *APITestSuite) TestListEnvironments() {
+	e1, err := types.NewEnvironment("e1", "td1", "cluster1")
+	e2, err := types.NewEnvironment("e2", "td2", "cluster2")
+	environments := []types.Environment{*e1, *e2}
+	suite.environment.EXPECT().ListEnvironments(gomock.Any()).Return(environments, nil)
+	request, err := http.NewRequest("GET", "/v1/environments", nil)
+	assert.Nil(suite.T(), err, "Unexpected error")
+
+	responseRecorder := httptest.NewRecorder()
+	suite.router.ServeHTTP(responseRecorder, request)
+
+	assert.Equal(suite.T(), http.StatusOK, responseRecorder.Code)
 
-	s.Path("/ping").
-		Methods("GET").
-		HandlerFunc(suite.api.Ping)
+	var environmentsModel models.Environments
+	b, _ := ioutil.ReadAll(responseRecorder.Body)
+	json.Unmarshal(b, &environmentsModel)
 
-	return s
+	for i := 0; i < len(environments); i++ {
+		environment := environments[i]
+		environmentModel := environmentsModel.Items[i]
+		suite.assertSame(&environment, environmentModel)
+	}
+}
+
+func (suite *APITestSuite) TestListEnvironmentsServerError() {
+	err := errors.New("Error when calling ListEnvironments")
+	suite.environment.EXPECT().ListEnvironments(gomock.Any()).Return(nil, err)
+	request, err := http.NewRequest("GET", "/v1/environments", nil)
+	assert.Nil(suite.T(), err, "Unexpected error")
+
+	responseRecorder := httptest.NewRecorder()
+	suite.router.ServeHTTP(responseRecorder, request)
+
+	assert.Equal(suite.T(), http.StatusInternalServerError, responseRecorder.Code)
+}
+
+func (suite *APITestSuite) TestDeleteEnvironment() {
+	name := "testEnv"
+	suite.environment.EXPECT().DeleteEnvironment(gomock.Any(), name).Return(nil)
+
+	request, err := http.NewRequest("DELETE", "/v1/environments/"+name, nil)
+	assert.Nil(suite.T(), err, "Unexpected error")
+
+	responseRecorder := httptest.NewRecorder()
+	suite.router.ServeHTTP(responseRecorder, request)
+
+	assert.Equal(suite.T(), http.StatusOK, responseRecorder.Code)
+}
+
+func (suite *APITestSuite) TestDeleteEnvironmentMissingEnvironment() {
+	name := "testEnv"
+	notfounderr := types.NewNotFoundError(errors.New("Environment is missing"))
+	suite.environment.EXPECT().DeleteEnvironment(gomock.Any(), name).Return(notfounderr)
+
+	request, err := http.NewRequest("DELETE", "/v1/environments/"+name, nil)
+	assert.Nil(suite.T(), err, "Unexpected error")
+
+	responseRecorder := httptest.NewRecorder()
+	suite.router.ServeHTTP(responseRecorder, request)
+
+	assert.Equal(suite.T(), http.StatusNotFound, responseRecorder.Code)
+}
+
+func (suite *APITestSuite) assertSame(environment *types.Environment, environmentModel *models.Environment) {
```

Comment:
It's just that this the API package and we may want to assert deployments later on. But we can deal with it when we have to later on I guess.